### ISSUE TITLE
Allow World Map Animations...

### ISSUE THREAD

**Issue by datahead8888** [Secretchronicles/TSC#483]
As discussed at http://forum.secretchronicles.de/topics/305 it would be great to be able to have animations for sprites in the world map.

To quote Quintus:
~~~~~~~~~~~~~~~~~~~~~~~~
There has never been a possibility to animate things on the world map. It’s a nice idea, but I guess this will be implemented as part of a more genaralised handling of animated tiles in TSC. Currently, each animated object requires some C++ code for it, whereas it’d be nicer to just have some loading code in a central class that reads in animation config files. Especially as we have the latter (.imgset files) already thanks to @simpletoon.

So long story short: Currently we can’t have animations on the world map.
~~~~~~~~~~~~~~~~~~~~~~~~

**Comment by brianvanderburg2** [comment]
I may take a look at this and see what can be done whenever I get some free time.  For now just some ideas and thoughts.

* overworld_loader.cpp:214 loads the sprites, calling sprite.cpp:318
* sprite.cpp:318 creates the sprite from the attributes, checks the image attribute, if it ends in ".png" it just uses Set_Image, otherwise it uses Add_Image_Set/Set_Image_Set.  This would allow for a sprite in the overworld file to have a filename given as ".imgset" instead, and the image set would be loaded and set.
* overworld.cpp:473 Update calls m_sprite_manager->Update_Items(), which should call the cSprite::Update, which in turn calls Update_Animation.

This seems like it should allow a sprite added to the overworld that would be animated as follows:

"""
<sprite>
   
   <property name="image" value="world/objects/hedge/green.imgset" />
</sprite>
"""

green.imgset:
"""
time 100
green_1.png
green_2.png
green_3.png
green_2.png
"""

Just from a quick skim of the code in devel, it seems like this should work, including plain sequential animations but also some randomness in frame branching.  Eventually I should add a file describing the .anim format.

**Comment by brianvanderburg2** [comment]
I just tested this by editing world 1's world.xml and setting some sprite to use an enemy .imgset file "enemy/larry/grey/walk.imgset".  It seems to work correctly.

**Comment by datahead8888** [comment]
Could you post it in a PR so that we can try it out?  Be sure to include testing instructions or worlds/levels as needed.

**Comment by brianvanderburg2** [comment]
Pull request here: #484

There isn't any changes to code needed, it just requires manually editing the world XML files.  Currently neither the level editor or world editor that I'm aware of have a way to show imgset items in the editor.  I don't know anything about the editor code so don't know how to change that.

**Comment by Quintus** [comment]
Very cool. Now we only need a to add a UI element in the editor that supports animated graphics...

Vale,
Quintus
